S3 upload to digitalocean 403  Forbidden

**Version:**
 - listmonk: 2.1.0
 - OS: macOS 10.15.7

**Description of the bug and steps to reproduce:**
Uploading S3 assets to digitalocean fails with 403 error:

```
error uploading file: status code: 403 Forbidden: "<?xml version=\"1.0\" encoding=\"UTF-8\"?><Error><Code>AccessDenied</Code><RequestId>tx000000000000028330f0d-00620d9368-2738c28d-nyc3c</RequestId><HostId>2738c28d-nyc3c-nyc3-zg03</HostId></Error>"
```

This happens since https://github.com/knadh/listmonk/pull/640. Reverting to `c.s3.FileUpload(p)` instead of `c.s3.FilePut(p)` fixes this issue.

Happy to open a PR with a change but I'm not very familiar with go convention and syntax. I understand the change to `FilePut` was to support B2  as mentioned in #617. Seems like `FileUpload()` is simples3's suggested function so what about something like this where listmonk tries `FileUpload` but tries to fall back to `FilePut` and then ultimately errors:

```go
// Upload.
if _, err := c.s3.FileUpload(p); err != nil {
    if _, err := c.s3.FilePut(p); err != nil {
        return "", err
    }
}
return name, nil
```

**Screenshots:**

Configuration:

<img width="1120" alt="image" src="https://user-images.githubusercontent.com/1012017/154379317-9772a5f2-e0f4-4147-9e50-943f0f807dbb.png">

I'm stuck. This is the output of `go test ./...`:

```
--- FAIL: TestS3_FileUploadPostAndPut (1.05s)
    --- FAIL: TestS3_FileUploadPostAndPut/Upload_test.txt_put (0.02s)
        simples3_test.go:172: S3.FileUpload() error = status code: 403 Forbidden: "<?xml version=\"1.0\" encoding=\"UTF-8\"?><Error><Code>AccessDenied</Code><RequestId>tx00000000000002906edb1-00620e673f-2738c28d-nyc3c</RequestId><HostId>2738c28d-nyc3c-nyc3-zg03</HostId></Error>", wantErr false
        simples3_test.go:180: S3.FileUpload() returned empty response, { map[]}
    --- FAIL: TestS3_FileUploadPostAndPut/Upload_test.txt_with_custom_metadata_post (0.09s)
        simples3_test.go:158: S3.FileUpload() error = status code: 403 Forbidden, wantErr false
        simples3_test.go:162: S3.FileDetails() returned incorrect metadata, got: simples3.DetailsResponse{ContentType:"", ContentLength:"", AcceptRanges:"", Date:"", Etag:"", LastModified:"", Server:"", AmzID2:"", AmzRequestID:"", AmzMeta:map[string]string(nil), ExtraHeaders:map[string]string(nil)}
    --- FAIL: TestS3_FileUploadPostAndPut/Upload_test.txt_with_custom_metadata_put (0.16s)
        simples3_test.go:172: S3.FileUpload() error = status code: 403 Forbidden: "<?xml version=\"1.0\" encoding=\"UTF-8\"?><Error><Code>AccessDenied</Code><RequestId>tx00000000000002909674c-00620e673f-2741fd34-nyc3c</RequestId><HostId>2741fd34-nyc3c-nyc3-zg03</HostId></Error>", wantErr false
        simples3_test.go:180: S3.FileUpload() returned empty response, { map[]}
        simples3_test.go:190: S3.FileUpload() error = status code: 403 Forbidden, wantErr false
        simples3_test.go:194: S3.FileDetails() returned incorrect metadata, got: simples3.DetailsResponse{ContentType:"", ContentLength:"", AcceptRanges:"", Date:"", Etag:"", LastModified:"", Server:"", AmzID2:"", AmzRequestID:"", AmzMeta:map[string]string(nil), ExtraHeaders:map[string]string(nil)}
    --- FAIL: TestS3_FileUploadPostAndPut/Upload_avatar.png_put (0.14s)
        simples3_test.go:172: S3.FileUpload() error = status code: 403 Forbidden: "<?xml version=\"1.0\" encoding=\"UTF-8\"?><Error><Code>AccessDenied</Code><RequestId>tx00000000000002906ee0d-00620e673f-2738c28d-nyc3c</RequestId><HostId>2738c28d-nyc3c-nyc3-zg03</HostId></Error>", wantErr false
        simples3_test.go:180: S3.FileUpload() returned empty response, { map[]}
    --- FAIL: TestS3_FileUploadPostAndPut/Upload_special_filename_txt_put (0.02s)
        simples3_test.go:172: S3.FileUpload() error = status code: 403 Forbidden: "<?xml version=\"1.0\" encoding=\"UTF-8\"?><Error><Code>AccessDenied</Code><RequestId>tx0000000000000290967e6-00620e6740-2741fd34-nyc3c</RequestId><HostId>2741fd34-nyc3c-nyc3-zg03</HostId></Error>", wantErr false
        simples3_test.go:180: S3.FileUpload() returned empty response, { map[]}
--- FAIL: TestS3_FileDownload (0.55s)
    --- FAIL: TestS3_FileDownload/txt (0.02s)
        simples3_test.go:292: S3.FileDownload() error = status code: 403 Forbidden, wantErr false
    --- FAIL: TestS3_FileDownload/png (0.32s)
        simples3_test.go:292: S3.FileDownload() error = status code: 403 Forbidden, wantErr false
    --- FAIL: TestS3_FileDownload/txt-special-filename (0.21s)
        simples3_test.go:292: S3.FileDownload() error = status code: 403 Forbidden, wantErr false
--- FAIL: TestS3_FileDelete (1.03s)
    --- FAIL: TestS3_FileDelete/Delete_test.txt (0.19s)
        simples3_test.go:390: S3.FileDelete() error = status code: 403 Forbidden, wantErr false
    --- FAIL: TestS3_FileDelete/Delete_test_metadata.txt (0.32s)
        simples3_test.go:390: S3.FileDelete() error = status code: 403 Forbidden, wantErr false
    --- FAIL: TestS3_FileDelete/Delete_avatar.png (0.19s)
        simples3_test.go:390: S3.FileDelete() error = status code: 403 Forbidden, wantErr false
    --- FAIL: TestS3_FileDelete/Delete_special_filename_txt (0.34s)
        simples3_test.go:390: S3.FileDelete() error = status code: 403 Forbidden, wantErr false
FAIL
FAIL	github.com/rhnvrm/simples3	5.352s
FAIL
```

--- a/go.mod
+++ b/go.mod
@@ -26,7 +26,7 @@ require (
 	github.com/niemeyer/pretty v0.0.0-20200227124842-a10e7caefd8e // indirect
 	github.com/paulbellamy/ratecounter v0.2.0 // indirect
 	github.com/pelletier/go-toml v1.9.4 // indirect
-	github.com/rhnvrm/simples3 v0.8.1
+	github.com/rhnvrm/simples3 v0.8.2
 	github.com/spf13/cast v1.4.1 // indirect
 	github.com/spf13/pflag v1.0.5
 	github.com/yuin/goldmark v1.4.1

